### DISCUSSION TITLE
Parse using a predefined tree sitter json grammar

### DISCUSSION THREAD

**Discussion by Alphapage** [zesterer/chumsky#655] [Q&A]
Hello,

I am trying to switch to Chumsky, but I would like to know if it is easy to translate from an existing tree sitter json file.
If you have any mapping for CHOICE,SEQ, REPEAT ... to apply to Chumsky where possible.
Maybe it doesn't use the same semantic.

Thank you in advance for your help.

**Comment by zesterer** [comment]
Equivalents of those all exist, yes: `choice`, `then`, `repeated`, etc. See the `primitive` module docs and the docs for the `Parser` trait.
